projects/graph/graph.py

```diff
@@ -75,27 +75,97 @@ def dft(self, starting_vertex):
                 for edge in self.vertices[node]:
                 ### put them on the stack
                     stack.push(edge)
-    def dft_recursive(self, starting_vertex):
+
+                    
+    def dft_recursive(self, node, visited=set()):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
         This should be done using recursion.
         """
-        pass  # TODO
+
+        #take the node (starting_vertex)
+        # if this node hasn't been visited:
+        if node not in visited:
+            print(node)
+        # mark it as visited
+            visited.add(node)
+        # for each of this node's neighbors:
+            for neighbor in self.vertices[node]:
+        # call dft_recursive
+                self.dft_recursive(neighbor, visited)
+
+      
     def bfs(self, starting_vertex, destination_vertex):
         """
         Return a list containing the shortest path from
         starting_vertex to destination_vertex in
         breath-first order.
         """
-        pass  # TODO
+        # make a queue
+        q = Queue()
+        # make a visted set
+        visited = set()
+        # enqueue the PATh to the node
+        q.enqueue([starting_vertex])
+        #while the queue isn't empty:
+        while q.size() > 0:
+        # dequeue the PATH
+            path = q.dequeue()
+        # the last thing in the path is our current item
+            node = path[-1]
+        # if it's not visited:
+            if node not in visited:
```

Comment:
I will place this condition in the for loop to prevent previously visited nodes from being added to the queue again.